app/components/spotlight/analytics/dashboard_component.html.erb

```diff
@@ -1,6 +1,27 @@
+<% if note %>
+  <p><%= note %></p>
+<% end %>
+
+<%= form_with url: request.path, class: 'mb-3 d-flex', method: :get do |form| %>
+  <div>
+    <%= form.label :start_date, I18n.t("spotlight.dashboards.analytics.start_date_label") %>
+    <%= form.date_field :start_date, value: dates['start_date'], min: min_date, max: max_date %>
+    <%= form.label :end_date, I18n.t("spotlight.dashboards.analytics.end_date_label"), class: 'ml-2 ms-2' %>
+    <%= form.date_field :end_date, value: dates['end_date'], min: min_date, max: max_date %>
+  </div>
+  <div class="ml-auto ms-auto">
+    <%= form.submit I18n.t("spotlight.dashboards.analytics.submit_date_label"), class: 'btn btn-primary' %>
+  </div>
+<% end %>
+
 <% if results? %>
 <%= cache current_exhibit, expires_in: 1.hour do %>
```

Comment:
With caching turned on in development the first request to the page gets cached and then none of the date range parameters I apply get reflected on the page. I think you need to change the cache key to account for any parameters applied.